CI failing with RuntimeError about no install candidates for drjit

### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

```

self = <conda_lock._vendor.poetry.installation.chooser.Chooser object at 0x7f10d1ded310>
package = Package('drjit', '0.4.3')

    def choose_for(self, package):  # type: (Package) -> Link
        """
        Return the url of the selected archive for a given package.
        """
        links = []
        for link in self._get_links(package):
            if link.is_wheel and not Wheel(link.filename).is_supported_by_environment(
                self._env
            ):
                continue
    
            if link.ext in {".egg", ".exe", ".msi", ".rpm", ".srpm"}:
                continue
    
            links.append(link)
    
        if not links:
>           raise RuntimeError(
                "Unable to find installation candidates for {}".format(package)
            )
E           RuntimeError: Unable to find installation candidates for drjit (0.4.3)
```

### Conda Info

_No response_

### Conda Config

_No response_

### Conda list

_No response_

### Additional Context

_No response_

--- a/tests/test_regression.py
+++ b/tests/test_regression.py
@@ -45,8 +45,9 @@ def test_pr_436(
         channels:
         - conda-forge
         dependencies:
+        - python 3.11
         - pip:
-            - drjit
+            - drjit 0.4.2
         """
     )
     (tmp_path / "environment.yml").write_text(spec)

--- a/tests/test_regression.py
+++ b/tests/test_regression.py
@@ -47,7 +47,7 @@ def test_pr_436(
         dependencies:
         - python 3.11
         - pip:
-            - drjit 0.4.2
+            - drjit==0.4.2
         """
     )
     (tmp_path / "environment.yml").write_text(spec)

